integrations/aws/utils.py

```diff
@@ -0,0 +1,256 @@
+    
+    :return: List of AWS account IDs.
+    """
+    logger.info("Updating AWS credentials")
+    aws_access_key_id = ocean.integration_config.get("aws_access_key_id")
+    aws_secret_access_key = ocean.integration_config.get("aws_secret_access_key")
+    if not aws_access_key_id or not aws_secret_access_key:
+        logger.error("Did not specify AWS account to use, please add aws user credentials")
+        return
+    
+    sts_client = boto3.client('sts', aws_access_key_id=aws_access_key_id, aws_secret_access_key=aws_secret_access_key)
+    caller_identity = sts_client.get_caller_identity()
+    current_account_id = caller_identity['Account']
+
+    _aws_credentials.append(AwsCredentials(
+        account_id=current_account_id,
+        access_key_id=aws_access_key_id,
+        secret_access_key=aws_secret_access_key,
+    ))
+
+    # TODO: change to be dynamic
+    ROLE_NAME = 'ocean-integ-poc-role'
+    organizations_client = sts_client.assume_role(
+        # TODO: change to be dynamic
+        RoleArn=f'arn:aws:iam::362207926288:role/AWS-Exporter-Ocean-POC-Organization-Role',
+        RoleSessionName='AssumeRoleSession'
+    )
+
+    credentials = organizations_client['Credentials']
+    # Get the list of all AWS accounts
+    organizations_client = boto3.client('organizations', 
+        aws_access_key_id=credentials['AccessKeyId'],
+        aws_secret_access_key=credentials['SecretAccessKey'],
+        aws_session_token=credentials['SessionToken']
+    )
+    paginator = organizations_client.get_paginator('list_accounts')
+    try:
+        for page in paginator.paginate():
+            for account in page['Accounts']:
+                try:
+                    account_role = sts_client.assume_role(
+                        RoleArn=f'arn:aws:iam::{account["Id"]}:role/{ROLE_NAME}',
+                        RoleSessionName='AssumeRoleSession'
+                    )
+                    credentials = account_role['Credentials']
+                    _aws_credentials.append(AwsCredentials(
+                        account_id=account['Id'],
+                        access_key_id=credentials['AccessKeyId'],
+                        secret_access_key=credentials['SecretAccessKey'],
+                        session_token=credentials['SessionToken']
+                    ))
+                except sts_client.exceptions.ClientError as e:
+                    # If assume_role fails due to permission issues or non-existent role, skip the account
+                    if e.response['Error']['Code'] == 'AccessDenied':
+                        continue
+                    else:
+                        raise
+    except organizations_client.exceptions.AccessDeniedException:
+        # If the caller is not a member of an AWS organization, assume_role will fail with AccessDenied
+        # In this case, assume the role in the current account
+        logger.error("Caller is not a member of an AWS organization. Assuming role in the current account.")
```

Comment:
worth logging the rolearn that you tried to assume role with

Reply:
I don't think it's best practice to show `arn`s or roles